src/mlpack/tests/knn_test.cpp

```diff
@@ -1766,7 +1766,51 @@ TEST_CASE("KNNGreedyTreeSearch", "[KNNTest]")
   // are present in dataset.
   REQUIRE(arma::accu(neighbors < 0 || neighbors >= 100) ==0);
 
-  // Check that all distances values are between 0.0 and 1.0 as arma::randu
+  // Check that all distances values are between 0.0 and sqrt(3) as arma::randu
   // generates a uniform distribution in [0, 1].
   REQUIRE(arma::accu(distances < 0.0 || distances > std::sqrt(3.0)) == 0);
 }
+
+/**
+ * Check that no garbage value is returned when greedy tree traversal
+ * is performed over spill trees.
+ */
+TEST_CASE("KNNSpillTreeSearchEnoughResults", "[KNNTest]")
+{
+  // Initalize dataset.
+  arma::mat dataset = arma::randu<arma::mat>(3, 100);
+
+  // Build tree with a leaf_size of 1.
```

Comment:
leaf_size, hm, should this be "leaf size", since this is no main test?

Reply:
Good point---fixed in 9fb1d3de2.